### PR TITLE
[ODLA] dnnl: bf16 full support for better performance

### PR DESCRIPTION
**Pull request by ahuizxc**
1. bf16 support for nms/topk ops
2. add new function erf_bf16_func
3. fix avx512 computation bugs
4. refactor gather_func code
5. bf16 mode, weights casting and packing.
6. full bf16 graph support

### CODE REVIEW

File: ODLA/platforms/dnnl/dnnl_utils_avx512.h [hunk 1]
```diff
@@ -144,14 +118,14 @@ inline void floorbf_func(int len, float* src, float* dst) {
     auto out1 = _mm512_floor_ps(a1);
```
Comment (weimingzha0) [inline ODLA/platforms/dnnl/dnnl_utils_avx512.h:118]: for line 112, should it be "for(; i < len - vec_size; i += vec_size)" ?

File: ODLA/platforms/dnnl/dnnl_utils_avx512.h [hunk 2]
```diff
@@ -144,14 +118,14 @@ inline void floorbf_func(int len, float* src, float* dst) {
...
 
     auto C_bf16 = _mm512_cvtne2ps_pbh(out1, out0);
-    _mm512_mask_storeu_ps(dst + i, mask16, _mm512_castsi512_ps(C_bf16));
+    _mm512_mask_storeu_ps(dst + i / 2, mask16, _mm512_castsi512_ps(C_bf16));
```
Comment (weimingzha0) [inline ODLA/platforms/dnnl/dnnl_utils_avx512.h:121]: should this still be "dst + i"?

Comment (ahuizxc) [reply ODLA/platforms/dnnl/dnnl_utils_avx512.h:121]: dst is float* and vec_size is 32, so we should store i/2 vec_size (512bit) in single loop :)

**Comment by weimingzha0** [comment]
LGTM

**Review by dj176050 — APPROVED** [review]

**Pull request merged**